src/pages/workspace/reimburse/WorkspaceReimburseNoVBAView.js

```diff
@@ -93,17 +84,14 @@ class WorkspaceReimburseNoVBAView extends React.Component {
     }
 
     setRate(value) {
-        const numValue = parseFloat(value);
-        if (Number.isNaN(numValue)) {
-            this.setState({rateValue: ''});
-            return;
-        }
-
-        // Set the immediate value so the user does not lose the input
-        this.setState({rateValue: numValue.toString()});
-
-        // Set the corrected value with a delay and sync to the server
-        this.updateRateValueThrottled(numValue);
+        const isInvalidRateValue = value.includes('.') && !CONST.REGEX.RATE_VALUE.test(value);
+
+        this.setState(prevState => ({
+            rateValue: !isInvalidRateValue ? value : prevState.rateValue,
+        }), () => {
+            // Set the corrected value with a delay and sync to the server
+            this.updateRateValueDebounced(this.state.rateValue);
+        });
```

Comment:
Debouncing is causing issues. It updates the values while the user is typing. May be related to calling it in setState completion callback.  But test it.

Reply:
It should be expected behavior.

Reply:
Debouncing is expected behaviour but we should not modify the value while user is updating it. RIght? So moving the cursor is also comes under updating. 

e.g. user types 123.1233| , moves the cursor back two times. He expects the cursor to be on 123.12|33 but as the value  changed to 123.12 cursor is now at 123.|12 which is bad UX.

Reply:
Yeah, Keydown looks like it but could you please test to see if increasing debouncing works.

Reply:
I like we start the debouncing on keydown to avoid having any issues while the user is trying to update the value.

Reply:
Thanks, Rocio. @kakajann you know what to do :arrow_up: . Thanks.

Reply:
Done